[processor/transform] Nil Pointer Reference Using With Prometheus Receiver

**Describe the bug**

Was getting this bug using the transform processor with the prometheus receiver. Appears to panic on a minimum of `convert_gauge_to_sum` and `set` queries however I believe there may be more.

I have also validated that this was not an issue in v0.58.0. I also validated that latest main appears to have this issue as well.

Error happens during the processing of metrics.

Error:

```
2022-09-06T13:28:00.100-0400	info	service/service.go:129	Everything is ready. Begin running and processing data.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x39e2879]

goroutine 147 [running]:
go.opentelemetry.io/collector/pdata/pmetric.Sum.DataPoints(...)
	go.opentelemetry.io/collector/pdata@v0.59.0/pmetric/generated_metrics.go:843
github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics.(*Processor).ProcessMetrics(0xc0008bf560?, {0x10?, 0x5dee3e0?}, {0xc001716c00?})
	github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor@v0.59.0/internal/metrics/processor.go:57 +0x259
go.opentelemetry.io/collector/processor/processorhelper.NewMetricsProcessor.func1({0x7ab8e58, 0xc0016244b0}, {0xc001716c00?})
	go.opentelemetry.io/collector@v0.59.0/processor/processorhelper/metrics.go:75 +0xf8
go.opentelemetry.io/collector/consumer.ConsumeMetricsFunc.ConsumeMetrics(...)
	go.opentelemetry.io/collector@v0.59.0/consumer/metrics.go:36
go.opentelemetry.io/collector/processor/processorhelper.NewMetricsProcessor.func1({0x7ab8e58, 0xc0016244b0}, {0x0?})
	go.opentelemetry.io/collector@v0.59.0/processor/processorhelper/metrics.go:83 +0x1dd
go.opentelemetry.io/collector/consumer.ConsumeMetricsFunc.ConsumeMetrics(...)
	go.opentelemetry.io/collector@v0.59.0/consumer/metrics.go:36
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal.(*transaction).Commit(0xc000f38000)
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver@v0.59.0/internal/transaction.go:195 +0x285
github.com/prometheus/prometheus/scrape.(*scrapeLoop).scrapeAndReport.func1()
	github.com/prometheus/prometheus@v0.37.0/scrape/scrape.go:1275 +0x45
github.com/prometheus/prometheus/scrape.(*scrapeLoop).scrapeAndReport(0xc00173e0e0, {0xedaa97a2c?, 0xb6b96a0?, 0x0?}, {0xedaa97a2c?, 0xb6b96a0?, 0xb6b96a0?}, 0x0)
	github.com/prometheus/prometheus@v0.37.0/scrape/scrape.go:1346 +0xf70
github.com/prometheus/prometheus/scrape.(*scrapeLoop).run(0xc00173e0e0, 0xc0013f99e0?)
	github.com/prometheus/prometheus@v0.37.0/scrape/scrape.go:1228 +0x345
created by github.com/prometheus/prometheus/scrape.(*scrapePool).sync
	github.com/prometheus/prometheus@v0.37.0/scrape/scrape.go:592 +0xa1e

```
**Steps to reproduce**
Use supplied config with 

**What did you expect to see?**
Processor should not have panicked

**What did you see instead?**
Panic with stacktrace.

**What version did you use?**
Version: v0.59.0, validated not an issue with v0.58.0

**What config did you use?**
Config: (e.g. the yaml config file)
```
```yaml
exporters:
  logging:
    loglevel: debug
processors:
  filter/couchbase_couchbase_1:
    metrics:
      exclude:
        match_type: strict
        metric_names:
        - scrape_samples_post_metric_relabeling
        - scrape_series_added
        - scrape_duration_seconds
        - scrape_samples_scraped
        - up
  transform/couchbase_couchbase_3:
    metrics:
      queries:
      - convert_gauge_to_sum("cumulative", true) where metric.name == "couchbase.bucket.error.oom.count"
      - set(metric.description, "Number of out of memory errors.") where metric.name == "couchbase.bucket.error.oom.count"
      - set(metric.unit, "{errors}") where metric.name == "couchbase.bucket.error.oom.count"
      - set(metric.description, "Number of items that belong to the bucket.") where metric.name == "couchbase.bucket.item.count"
      - set(metric.unit, "{items}") where metric.name == "couchbase.bucket.item.count"
      - convert_gauge_to_sum("cumulative", true) where metric.name == "couchbase.bucket.item.ejection.count"
      - set(metric.description, "Number of item value ejections from memory to disk.") where metric.name == "couchbase.bucket.item.ejection.count"
      - set(metric.unit, "{ejections}") where metric.name == "couchbase.bucket.item.ejection.count"
      - set(metric.description, "The memory usage at which items will be ejected.") where metric.name == "couchbase.bucket.memory.high_water_mark.limit"
      - set(metric.unit, "By") where metric.name == "couchbase.bucket.memory.high_water_mark.limit"
      - set(metric.description, "The memory usage at which ejections will stop that were previously triggered by a high water mark breach.") where metric.name == "couchbase.bucket.memory.low_water_mark.limit"
      - set(metric.unit, "By") where metric.name == "couchbase.bucket.memory.low_water_mark.limit"
      - set(metric.description, "Usage of total memory available to the bucket.") where metric.name == "couchbase.bucket.memory.usage"
      - set(metric.unit, "By") where metric.name == "couchbase.bucket.memory.usage"
      - convert_gauge_to_sum("cumulative", true) where metric.name == "couchbase.bucket.operation.count"
      - set(metric.description, "Number of operations on the bucket.") where metric.name == "couchbase.bucket.operation.count"
      - set(metric.unit, "{operations}") where metric.name == "couchbase.bucket.operation.count"
      - set(metric.description, "Number of non-resident vBuckets.") where metric.name == "couchbase.bucket.vbucket.count"
      - set(metric.unit, "{vbuckets}") where metric.name == "couchbase.bucket.vbucket.count"
receivers:
  prometheus/couchbase_couchbase:
    config:
      scrape_configs:
      - basic_auth:
          password: otelpassword
          username: otelu
        job_name: couchbase
        metric_relabel_configs:
        - action: keep
          regex: (kv_ops)|(kv_vb_curr_items)|(kv_num_vbuckets)|(kv_total_memory_used_bytes)|(kv_ep_num_value_ejects)|(kv_ep_mem_high_wat)|(kv_ep_mem_low_wat)|(kv_ep_oom_errors)
          source_labels:
          - __name__
        scrape_interval: 10s
        static_configs:
        - targets:
          - localhost:8091
service:
  pipelines:
    metrics/couchbase_couchbase:
      exporters:
      - logging
      processors:
      - filter/couchbase_couchbase_1
      - transform/couchbase_couchbase_3
      receivers:
      - prometheus/couchbase_couchbase

```

**Environment**
OS: Darwin && Debian 10
Compiler(if manually compiled): released version v0.59.0 and compiled version using go 1.18.5

**Additional context**

Let me know if there may be a user error or missed changelog but as far as I can tell things should work as previous.

--- a/processor/transformprocessor/internal/metrics/processor.go
+++ b/processor/transformprocessor/internal/metrics/processor.go
@@ -54,7 +54,7 @@ func (p *Processor) ProcessMetrics(_ context.Context, td pmetric.Metrics) (pmetr
 				case pmetric.MetricDataTypeSum:
 					p.handleNumberDataPoints(metric.Sum().DataPoints(), metrics.At(k), metrics, smetrics.Scope(), rmetrics.Resource())
 				case pmetric.MetricDataTypeGauge:
-					p.handleNumberDataPoints(metric.Sum().DataPoints(), metrics.At(k), metrics, smetrics.Scope(), rmetrics.Resource())
+					p.handleNumberDataPoints(metric.Gauge().DataPoints(), metrics.At(k), metrics, smetrics.Scope(), rmetrics.Resource())
 				case pmetric.MetricDataTypeHistogram:
 					p.handleHistogramDataPoints(metric.Histogram().DataPoints(), metrics.At(k), metrics, smetrics.Scope(), rmetrics.Resource())
 				case pmetric.MetricDataTypeExponentialHistogram:

--- a/unreleased/transform-fix-panic-gauge-type.yaml
+++ b/unreleased/transform-fix-panic-gauge-type.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: transformprocessor
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fixes panic of transformprocessor handling Gauge types
+
+# One or more tracking issues related to the change
+issues: [13905]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

